source/maths/mtx_transpose.c

```diff
@@ -0,0 +1,15 @@
+#include <c3d/maths.h>
+
+void Mtx_Transpose(C3D_Mtx* out)
+{
+	float swap;
+	for (int i = 0; i <= 2; i++)
```

Comment:
Yes, each `for` loop runs 3 iterations, `0`, `1`, and `2`. It does not run `3` because this is optimized to leave out unnecessary calculations. The `i` and `j` are calculating the rows/columns, not the values.

The first row contains 3 columns that need to be swapped out to the last 3 rows in the fourth column.
The second row contains 2 columns that need to be swapped out to the last 2 rows in the third column.
The third and final row contains 1 column that needs to be swapped out to the fourth row in the second column.
The swapping does not touch the matrix diagonal, therefore the algorithm doesn't extend beyond a total of 6 swaps, hence the total iterations of the `for` loops is 6.

And that is it. If you break 4x4 matrix transposing down, you're only doing 6 swaps, instead of 10 swaps with 4 unnecessary swaps included.